site/source/docs/porting/pthreads.rst

```diff
@@ -43,17 +43,82 @@ but is unrelated. That flag does not use pthreads or SharedArrayBuffer, and
 instead uses a plain Web Worker to run your main program (and postMessage to
 proxy messages back and forth).
 
+Proxying
+========
+
+The Web allows certain operations to only happen from the main browser thread,
+like interacting with the DOM. As a result, various operations are proxied to
+the main browser thread if they are called on a background thread. See
+`bug 3495 <https://github.com/emscripten-core/emscripten/issues/3495>`_ for
+more information and how to try to work around this until this. To check which
+operations are proxied, you can look for the function's implementation in
+the JS library (``src/library_*``) and see if it is annotated with
+``__proxy: 'sync'`` or ``__proxy: 'async'``; however, note that the browser
+itself proxies certain things (like some GL operations), so there is no
+general way to be safe here (aside from not blocking on the main browser
+thread).
+
+In addition, Emscripten currently has a simple model of file I/O only happening
+on the main application thread (as we support JS plugin filesystems, which
+cannot share memory), which is another set of operations that are proxied.
+
+Normally you don't need to worry about this proxying, as it is fully automatic,
+however see the section on blocking right after this one.
+
+Blocking on the main browser thread
+===================================
+
+Note that in most cases the "main browser thread" is the same as the "main
+application thread". The main browser thread is literally the browser's main
+thread where DOM access is allowed, and the main application thread is the one
+on which you started up the application. If you started it on the main browser
```

Comment:
how about "on which you started up the application by instantiation the emscripten Module object"?

Reply:
I'm not sure what you mean by "instantiating the Emscripten Module object"?

How about `The main application thread is the one on which you started up the Emscripten JavaScript file`?

Reply:
"Or laoded the emscripten JavaScript file"? I mean, there isn't "the" emscripten JS file, there's a different one for every binary, and it defines the `Module` object, and whichever thread/VM instance has that object is the application main thread.

Reply:
What worries me is that some users may not even be aware of the Module object. If emcc just emits your html and that runs your program, you don't need to interact with it.

Maybe `[..] started up the application (by loading the main JS file emitted by Emscripten)`?

Reply:
Yeah, I like that.